### ISSUE TITLE
OCaml multi line string detection

### ISSUE THREAD

**Issue by Pitometsu** [Fuco1/smartparens#988]
```ocaml
{example|
sting with
mulitiple lines
|example}
```
It does not detected as parenthesis by smartparens.

**Comment by Pitometsu** [comment]
where `{id|` is begin and `|id}` is end of the expression.

**Comment by Pitometsu** [comment]
See: https://github.com/ocaml/tuareg/issues/24

**Comment by Fuco1** [comment]
I don't think this is currently easily possible.  We could write some code for `{}` pair to check if there is `id|` and `|id` at the inner borders of the sexp and adjust the returned structure.

**Comment by Pitometsu** [comment]
Looks like smartparents in general not very compatible with OCaml mode, even comments like
`(** Comment. *)` it ignoring.

**Comment by Pitometsu** [comment]
Another cases to check are some "ppx" expressions:
`[@@deriving protocol ~driver:(module Jsonm)]`
`[%of_sexp: t]`
